students/rhamersky/Lesson_10/mailroom_part6.py

```diff
@@ -0,0 +1,392 @@
+        while True:
+            # Appends the donor list.
+            if donor_name == "List":
+                # --> Verifies that there is at least a donor is stored as a list row in the table tuple.
+                if len(dic_donors) > 0:
+                    Data.print_donor_names()  # --> Prints a list of donors
+                    break
+                else:
+                    print(
+                        "\n" + "There are no donors in list.")  # --> Let's the user know that a donor list does not exist.
+                    break
+
+            try:
+                flo_donation_amount = float(input(str_name + ", please enter donation amount. "))
+
+                # Updates donor tuple
+                new_donor = Donor(donor_name,flo_donation_amount)
+                dic_donors = Data.update_donor(new_donor.donor, new_donor.donation, new_donor.initial_donation)
+
+                # String formatting for a brief thank-you to donor.
+                thankyou_message = "\n"+"{}, thank you for donating ${:,.2f}.".format(donor_name, flo_donation_amount)
+                print(thankyou_message)
+                break
+
+            except ValueError:  # --> Allows the program to continue if the user does not enter the right data type.
+                print(str_name + ", not a valid amount. \n")
+                continue
+
+    @staticmethod
+    def user_donor_input():
+        '''
+        Try and except block used on User's option choice to check for integers.
+        :return: Returns donor name to main function.
+        '''
+        while True:
+            try:
+                str_user_input = input(str_name + ", please enter a donor name to do a projection on. ").title()
+            except ValueError:
+                print(str_name + ", not a valid option. \n")
+                # Verifies that the data type is part of the string class and that the donor name is in the donor dic.
+            else:
+                if str_user_input.__class__ == str().__class__ and str_user_input in dic_donors:
+                    str_donations = "Yes"
+                    break
+                else:
+                    print("Donor is not in donor list. \n")
+        return str_user_input, str_donations
+
+    @staticmethod
+    def user_factor_input():
+        '''
+        Try and except block used on User's option choice to check for integers.
+        :return: Returns factor input to the main function.
+        '''
+        while True:
+            try:
+                int_user_input = float(input(str_name + ", please enter a positive integer. "))
+            except ValueError:
+                print(str_name + ", not a valid option. \n")
+                # Verifies that the data type is part of the int class and that the user input a value greater than 0.
+            else:
+                if int_user_input.__class__ == float().__class__ and int_user_input > 0:
```

Comment:
Don't need to check for `float` here since you converted in the `try` block and you will never reach this `else` clause unless the value was successfully converted to a `float`.